activerecord/test/cases/defaults_test.rb

```diff
@@ -275,3 +263,17 @@ def with_mysql_not_null_table
     end
   end
 end
+
+if current_adapter?(:SQLite3Adapter)
+  class Sqlite3DefaultExpressionTest < ActiveRecord::TestCase
+    include SchemaDumpingHelper
+
+    test "schema dump includes default expression" do
+      output = dump_table_schema("defaults")
+      assert_match %r/t\.date\s+"modified_date",\s+default: -> { "CURRENT_DATE" }/, output
+      assert_match %r/t\.datetime\s+"modified_time",\s+default: -> { "CURRENT_TIMESTAMP" }/, output
+      assert_match %r/t\.datetime\s+"modified_time_without_precision",\s+precision: 0,\s+default: -> { "CURRENT_TIMESTAMP" }/, output
```

Comment:
Should this line be dumped as `precision: nil` instead of `precision: 0`?

https://github.com/rails/rails/blob/28292dc61f44e2b45bba94af5c44dc20e82ea197/activerecord/test/schema/sqlite_specific_schema.rb#L9

Reply:
Not really. Unless `nil` and `0` have different meaning for the database, which I don't think is the case.

Reply:
I think I can keep `nil` though, so I'll do that.

Reply:
I think MySQL is the only case that datetime without precision (i.e. `precision: nil`) truncates subseconds,
All others (PostgreSQL, SQLite3 (on the adapter in our codebase), probably SQLServer, Oracle) doesn't truncate subseconds, so as long as except MySQL, I think `nil` and `0` have different meaning for the supported databases.

Reply:
Updated

Reply:
Evidence for PostgreSQL:

```
# create table precision_test (ts_precision_nil timestamp, ts_precision_zero timestamp(0));
CREATE TABLE

# \d+ precision_test
                                               Table "public.precision_test"
      Column       |              Type              | Collation | Nullable | Default | Storage | Stats target | Description
-------------------+--------------------------------+-----------+----------+---------+---------+--------------+-------------
 ts_precision_nil  | timestamp without time zone    |           |          |         | plain   |              |
 ts_precision_zero | timestamp(0) without time zone |           |          |         | plain   |              |

# insert into precision_test values (now(), now());
INSERT 0 1

# select * from precision_test;
      ts_precision_nil      |  ts_precision_zero  
----------------------------+---------------------
 2022-02-08 10:30:14.275916 | 2022-02-08 10:30:14
(1 row)